File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: str
    company_id: str
    user_id: str
    title: str
    message: str
    type: str
    category: str | None = None
    link: str | None = None
    is_read: bool
    actor_id: str | None = None
    actor_name: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    page: int
    per_page: int
    unread_count: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  company_id: string;
  user_id: string;
  title: string;
  message: string;
  type: "info" | "success" | "warning" | "error";
  category: string | null;
  link: string | null;
  is_read: boolean;
  actor_id: string | null;
  actor_name: string | null;
  created_at: string;
}

export interface PaginatedNotifications {
  items: Notification[];
  total: number;
  page: number;
  per_page: number;
  unread_count: number;
}
